src/engine/GroupBy.h

```diff
@@ -158,13 +158,18 @@ class GroupBy : public Operation {
 
   // Data to perform the AVG aggregation using the HashMap optimization.
   struct AverageAggregationData {
+    using ValueGetter = sparqlExpression::detail::NumericValueGetter;
     bool error_ = false;
     double sum_ = 0;
     int64_t count_ = 0;
-    void increment(sparqlExpression::detail::NumericValue intermediate) {
-      if (const int64_t* intval = std::get_if<int64_t>(&intermediate))
+    void increment(auto&& value,
+                   const sparqlExpression::EvaluationContext* ctx) {
+      auto valueGetter = ValueGetter{};
+      auto val = valueGetter(AD_FWD(value), ctx);
```

Comment:
```suggestion
      auto val = ValueGetter{}(AD_FWD(value), ctx);
```

No need for the extra line
(same for the other getter below).